plenum/server/primary_elector.py

```diff
@@ -179,13 +186,20 @@ def scheduleElection(self):
 
     def startElection(self):
         """
-        Start the election process by nominating self as primary.
+        Start the election process by nominating self as primary. Calling this
+        will trigger the election for all instances, to trigger election for
+        only one instance, use `start_election_for_instance`
         """
         logger.debug("{} starting election".format(self))
         for r in self.replicas:
             self.prepareReplicaForElection(r)
 
-        self.nominateItself()
+        self._schedule(self.nominateItself, random.random())
+
+    def start_election_for_instance(self, inst_id):
+        # Called when starting election for a particular protocol instance
+        self.prepareReplicaForElection(self.replicas[inst_id])
+        self._schedule(self.nominateItself, random.random())
```

Comment:
Anytime we have an event in system, like start of election or a re-election where multiple nodes will nominate themselves we do a random wait so elections can terminate quickly